src/board_controller/pieeg/pieeg_board.cpp

```diff
@@ -0,0 +1,364 @@
+#include <string.h>
+#include <vector>
+
+#include "custom_cast.h"
+#include "math.h"
+#include "pieeg_board.h"
+#include "timestamp.h"
+
+#ifdef USE_PERIPHERY
+
+PIEEGBoard::PIEEGBoard(int board_id, struct BrainFlowInputParams params) : Board(board_id, params)
+{
+    spi = NULL;
+    gpio_in = NULL;
+    keep_alive = false;
+    initialized = false;
+    server_socket = nullptr;
+}
+
+PIEEGBoard::~PIEEGBoard()
+{
+    skip_logs = true;
+    release_session();
+}
+
+int PIEEGBoard::prepare_session()
+{
+    if (initialized)
+    {
+        safe_logger(spdlog::level::info, "Session already prepared");
+        return (int)BrainFlowExitCodes::STATUS_OK;
+    }
+    if (params.serial_port.empty())
+    {
+        params.serial_port = "/dev/spidev0.0";
+        safe_logger(spdlog::level::info, "Use serial port {}", params.serial_port.c_str());
+    }
+
+    gpio_in = gpio_new();
+    if (gpio_open(gpio_in, "/dev/gpiochip0", 26, GPIO_DIR_IN) < 0)
+    {
+        safe_logger(spdlog::level::err, "failed to open gpio");
+        gpio_free(gpio_in);
+        gpio_in = NULL;
+        return (int)BrainFlowExitCodes::UNABLE_TO_OPEN_PORT_ERROR;
+    }
+    spi = spi_new();
+    if (spi_open_advanced(spi, params.serial_port.c_str(), 0b01, 1000000, MSB_FIRST, 8, 1) < 0)
+    {
+        safe_logger(spdlog::level::err, "failed to open spi dev");
+        spi_free(spi);
+        spi = NULL;
+        gpio_free(gpio_in);
+        gpio_in = NULL;
+        return (int)BrainFlowExitCodes::UNABLE_TO_OPEN_PORT_ERROR;
+    }
+    int gpio_res = gpio_set_edge(gpio_in, GPIO_EDGE_FALLING);
+    if (gpio_res != 0)
+    {
+        safe_logger(spdlog::level::err, "failed to set gpio edge event handler: {}", gpio_res);
+        spi_free(spi);
+        spi = NULL;
+        gpio_free(gpio_in);
+        gpio_in = NULL;
+        return (int)BrainFlowExitCodes::UNABLE_TO_OPEN_PORT_ERROR;
+    }
+
+    server_socket = new SocketServerTCP("0.0.0.0", params.ip_port, true);
```

Comment:
what is it?

Reply:
Removed the socket server initialization as data streaming should not be handled here.